packages/labs/virtualizer/src/test/scenarios/element-and-directive-parity.test.ts

```diff
@@ -148,6 +148,11 @@ describe('lit-virtualizer and virtualize directive', () => {
     expect(ulv.shadowRoot?.textContent).to.include('[5 selected]');
     expect(uvd.shadowRoot?.textContent).to.include('[5 selected]');
 
+    // There may occasionally be visibility change events from the initial render, so
+    // we clear those out before we start watching for new ones to avoid flaky tess.
+    ulv.visibilityChangedEvents.splice(0);
```

Comment:
Do we expect at most one event here? Should we validate that in addition to clearing to ensure we don't have some other unexpected state creep in?

And when we clear, would setting `length = 0` have a more obvious intention?

Reply:
I'm also seeing we clear these again in lines 182-183 before the assertion happens. Could the flakiness be somewhere in between those?

Reply:
@justinfagnani: 
> And when we clear, would setting length = 0 have a more obvious intention?

So I was today years old when I learned you can set length to a smaller value in order to truncate arrays in JavaScript.  Fixed.

@augustjk:
While there's a non-zero possibility that more events could come between the time that the `await until(...)` finds a greater than 0 length on the event collector, I think the most likely cause for flakes was related to behavior during the settling phase of initial virtualizer render.